### PR TITLE
update get evaluations and send reminder apis

### PR DESCRIPTION
**Pull request by emmaacha**
FE PR: [add evaluation admin progress screen #208
](https://github.com/nerubia/kh360-react/pull/208)

### CODE REVIEW

File: src/services/evaluation-administration-service.ts [hunk 1]
```diff
@@ -234,7 +234,73 @@ export const close = async (id: number) => {
...
+  if (emailTemplate === null) {
+    throw new CustomError("Template not found", 400)
+  }
+
+  const scheduleEndDate = format(
+    evaluationAdministration.eval_schedule_end_date ?? new Date(),
+    "EEEE, MMMM d, yyyy"
+  )
+
+  const evaluation = await EvaluationRepository.getByFilters({
```
Comment (jatnerubia) [inline src/services/evaluation-administration-service.ts:257]: Does it work for external users? 🤔

**Review by jatnerubia — CHANGES_REQUESTED** [review]
^^

File: src/services/evaluation-administration-service.ts [hunk 2]
```diff
@@ -234,7 +234,91 @@ export const close = async (id: number) => {
...
+  const evaluation = await EvaluationRepository.getByFilters(filter)
+
+  const evaluator =
+    evaluation?.is_external === true
+      ? await ExternalUserRepository.getById(evaluation.external_evaluator_id ?? 0)
+      : await UserRepository.getById(evaluation?.evaluator_id ?? 0)
+  if (evaluator !== null) {
+    const evaluations = await EvaluationRepository.getAllByFilters({
+      evaluation_administration_id: evaluationAdministration.id,
+      evaluator_id: evaluator.id,
```
Comment (jatnerubia) [inline src/services/evaluation-administration-service.ts:291]: `evaluator_id` is for internal user 🤔

**Review by jatnerubia — CHANGES_REQUESTED** [review]
^^

File: src/services/evaluation-administration-service.ts [hunk 3]
```diff
@@ -234,7 +234,84 @@ export const close = async (id: number) => {
   )
 }
 
-export const sendReminder = async (id: number) => {
+export const sendReminderByEvaluator = async (id: number, evaluator_id: number, email: string) => {
```
Comment (jatnerubia) [inline src/services/evaluation-administration-service.ts:237]: I think we should pass `user_id` & `is_external` here 🤔

File: src/services/evaluation-administration-service.ts [hunk 4]
```diff
@@ -234,7 +234,84 @@ export const close = async (id: number) => {
...
+  if (emailTemplate === null) {
+    throw new CustomError("Template not found", 400)
+  }
+
+  const scheduleEndDate = format(
+    evaluationAdministration.eval_schedule_end_date ?? new Date(),
+    "EEEE, MMMM d, yyyy"
+  )
+
+  const externalEvaluator = await ExternalUserRepository.getByFilters({
```
Comment (jatnerubia) [inline src/services/evaluation-administration-service.ts:257]: then we can remove this

File: src/services/evaluation-administration-service.ts [hunk 5]
```diff
@@ -234,7 +234,84 @@ export const close = async (id: number) => {
...
+  })
+
+  const isExternal: boolean = externalEvaluator !== null
+
+  const filter = {
+    evaluation_administration_id: evaluationAdministration.id,
+    for_evaluation: true,
+    status: {
+      in: [EvaluationStatus.Open, EvaluationStatus.Ongoing],
+    },
```
Comment (jatnerubia) [inline src/services/evaluation-administration-service.ts:269]: and just add `...(is_external ? { external_evaluator_id: user_id } : { evaluator_id: user_id })` here

**Review by jatnerubia — CHANGES_REQUESTED** [review]
^^ also update `getEvaluators` to include `is_external`

File: src/services/evaluation-administration-service.ts (hunk 3 above)
Comment (emmaacha) [reply src/services/evaluation-administration-service.ts:237]: Is `user_id` the same with `evaluator_id` kuys @jatnerubia ?

Comment (jatnerubia) [reply src/services/evaluation-administration-service.ts:237]: Yes

**Review by jatnerubia — APPROVED** [review]
Cool

**Review by Dev-Nino — APPROVED** [review]

**Pull request merged**